File: uc/computerScience2/fall2013/liveDemos/week03/pizza-04/pizza.cpp
#include "pizza.h"


Pizza::Pizza() {
  // Nothing to do
}

void Pizza::addTopping(Topping topping) {
  toppings.push_back(topping);
}

vector<Topping> Pizza::getToppings() {
  return toppings;
}

int Pizza::getToppingsCount() {
  return toppings.size();
}

double Pizza::getPrice() {
  double price = 0.0;
  for (vector<Topping>::iterator it = toppings.begin() ;
      it != toppings.end(); it++) {
    if (it->isPremium()) {
      price += 1.25;
    } else {
      price += 0.75;
    }
  }
  return price;
}

File: uc/computerScience2/spring2014/liveDemos/java/pizza/src/edu/uc/eecs/pizza/Pizza.java
package edu.uc.eecs.pizza;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Pizza {
    public static final double BASE_PRICE = 8.99;
  private static int pizzaCount = 0;
  
  private List<Topping> toppings = new SortedList<Topping>();
  
  public Pizza() {
    pizzaCount++; // increment static counter
  }

  public void addTopping(Topping topping) {
    toppings.add(topping);
  }

  public List<Topping> getToppings() {
    return new ArrayList<Topping>(toppings);
  }
  
  public List<Topping> getToppingsPremiumFirst() {
    List<Topping> toppins = new ArrayList<Topping>(toppings);
    Collections.sort(toppins, new ToppingPremiumComparator());
    return toppins;
  }

  public int getToppingsCount() {
    return toppings.size();
  }
  
  /**
   * Calculates the prize of the pizza with the currently selected toppings.
   * Regular toppings att $0.75 to the price.
   * Premium toppings att $1.25 to the price.
   */
  public double getPrice() {
    double price = BASE_PRICE;
    for (Topping topping : toppings) {
      if (topping.isPremium()) {
        price += 1.25;
      } else {
        price += 0.75;
      }
    }
    return price;
  }

  public static int getTotalPizzaCount() {
    return pizzaCount;
  }
}
